client/asset/interface.go

```diff
@@ -290,20 +297,81 @@ type Order struct {
 	Immediate bool
 }
 
-// OrderEstimate is an estimate of the fees and locked amounts associated with
+// SwapEstimate is an estimate of the fees and locked amounts associated with
 // an order.
-type OrderEstimate struct {
+type SwapEstimate struct {
 	// Lots is the number of lots in the order.
-	Lots uint64
+	Lots uint64 `json:"lots"`
 	// Value is the total value of the order.
-	Value uint64
+	Value uint64 `json:"value"`
 	// MaxFees is the maximum possible fees that can be assessed for the order's
 	// swaps.
-	MaxFees uint64
-	// EstimatedFees is an estimation of the fees that might be assessed for the
-	// order's swaps.
-	EstimatedFees uint64
+	MaxFees uint64 `json:"maxFees"`
+	// RealisticWorstCase is an estimation of the fees that might be assessed in
+	// a worst-case scenario of 1 tx per 1 lot match, but at the prevailing fee
+	// rate estimate.
+	RealisticWorstCase uint64 `json:"realisticWorstCase"`
+	// RealisticBestCase is an estimation of the fees that might be assessed in
+	// a best-case scenario of 1 tx and 1 output for the entire order.
+	RealisticBestCase uint64 `json:"realisticBestCase"`
 	// Locked is the amount that will be locked if this order is
 	// subsequently placed.
-	Locked uint64
+	Locked uint64 `json:"locked"`
+}
+
+// RedeemEstimate is an estimate of the range of fees that might realistically
+// be assessed to the redemption transaction.
+type RedeemEstimate struct {
+	// RealisticBestCase is the best-case scenario fees of a single transaction
+	// with a match covering the entire order, at the prevailing fee rate.
+	RealisticBestCase uint64 `json:"realisticBestCase"`
+	// RealisticWorstCase is the worst-case scenario fees of all 1-lot matches,
+	// each with their own call to Redeem.
+	RealisticWorstCase uint64 `json:"realisticWorstCase"`
+}
+
+// PreSwapForm can be used to get a swap fees estimate.
+type PreSwapForm struct {
+	// LotSize is the lot size for the calculation. For quote assets, LotSize
+	// will can be based on either the user's limit order rate, or some measure
+	// of the current market conditions.
+	LotSize uint64
+	// Lots is the number of lots in the order.
+	Lots uint64
+	// AssetConfig is the dex's asset configuration info.
+	AssetConfig *dex.Asset
+	// Immediate should be set to true if this is for an order that is not a
+	// standing order, likely a market order or a limit order with immediate
+	// time-in-force.
+	Immediate bool
+}
+
+// SwapOption is an OrderEstimate and it's associated ConfigOption.
+type SwapOption struct {
+	ConfigOption
+	// Estimate is the OrderEstimate for an order with the specified
+	// option values.
+	Estimate *SwapEstimate
+}
+
+// PreSwap is a SwapEstimate returned from Wallet.PreSwap. The struct will be
+// expanded in in-progress work to accomondate order-time options.
+type PreSwap struct {
+	Estimate *SwapEstimate `json:"estimate"`
+}
```

Comment:
This is one of a few seemingly unnecessary wrapper structs that will soon be used to accommodate the order-time options. I've made some temporary comments in this file on the to explain their purpose.

Since order-time options are entirely optional, setting up the structs now means that when the options become available, we don't need to break the API.